server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java

```diff
@@ -0,0 +1,766 @@
+import org.apache.accumulo.core.metadata.schema.ExternalCompactionId;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata.LocationType;
+import org.apache.accumulo.core.rpc.ThriftUtil;
+import org.apache.accumulo.core.securityImpl.thrift.TCredentials;
+import org.apache.accumulo.core.tabletserver.thrift.TCompactionQueueSummary;
+import org.apache.accumulo.core.tabletserver.thrift.TCompactionStats;
+import org.apache.accumulo.core.tabletserver.thrift.TExternalCompactionJob;
+import org.apache.accumulo.core.tabletserver.thrift.TabletClientService;
+import org.apache.accumulo.core.trace.TraceUtil;
+import org.apache.accumulo.core.trace.thrift.TInfo;
+import org.apache.accumulo.core.util.HostAndPort;
+import org.apache.accumulo.core.util.compaction.ExternalCompactionUtil;
+import org.apache.accumulo.core.util.threads.ThreadPools;
+import org.apache.accumulo.fate.util.UtilWaitThread;
+import org.apache.accumulo.fate.zookeeper.ServiceLock;
+import org.apache.accumulo.fate.zookeeper.ZooReaderWriter;
+import org.apache.accumulo.server.AbstractServer;
+import org.apache.accumulo.server.GarbageCollectionLogger;
+import org.apache.accumulo.server.ServerOpts;
+import org.apache.accumulo.server.compaction.RetryableThriftCall;
+import org.apache.accumulo.server.compaction.RetryableThriftCall.RetriesExceededException;
+import org.apache.accumulo.server.compaction.RetryableThriftFunction;
+import org.apache.accumulo.server.manager.LiveTServerSet;
+import org.apache.accumulo.server.manager.LiveTServerSet.TServerConnection;
+import org.apache.accumulo.server.rpc.ServerAddress;
+import org.apache.accumulo.server.rpc.TCredentialsUpdatingWrapper;
+import org.apache.accumulo.server.rpc.TServerUtils;
+import org.apache.accumulo.server.rpc.ThriftServerType;
+import org.apache.accumulo.server.security.AuditedSecurityOperation;
+import org.apache.accumulo.server.security.SecurityOperation;
+import org.apache.thrift.TException;
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportException;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CompactionCoordinator extends AbstractServer
+    implements org.apache.accumulo.core.compaction.thrift.CompactionCoordinator.Iface,
+    LiveTServerSet.Listener {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompactionCoordinator.class);
+  private static final long TIME_BETWEEN_GC_CHECKS = 5000;
+  private static final long FIFTEEN_MINUTES =
+      TimeUnit.MILLISECONDS.convert(Duration.of(15, TimeUnit.MINUTES.toChronoUnit()));
+
+  protected static final QueueSummaries QUEUE_SUMMARIES = new QueueSummaries();
+
+  /* Map of compactionId to RunningCompactions */
+  protected static final Map<ExternalCompactionId,RunningCompaction> RUNNING =
```

Comment:
Maybe the dead compaction detector should remove things from here when it finds a dead external compaction.

Reply:
I wonder if the DeadCompactionDetector should have a reference to the CompactionCoordinator and call `compactionFailed` instead of calling the CompactionFinalizer directly, that way the behavior and logic are the same as a normal failure.

Reply:
That could work w/ some slight tweaks.  The compactionFailed method in the coordinator is kinda thrifty and only takes a single ecid.  Could have package private non thrifty compactionFailed method in the coordinator that takes a collections ecids and is called by the dead compaction detector and the thrifty compactionFailed method.

Reply:
I'm going to look at making this change.

Reply:
@keith-turner - I resolved this in f39707a492aea982b5127c7358e4a7f209a812f0